src/qt/listctrl.cpp

```diff
@@ -871,8 +871,32 @@ long wxListCtrl::DoInsertColumn(long col, const wxListItem& info)
 }
 
 
-void wxListCtrl::SetItemCount(long WXUNUSED(count))
+void wxListCtrl::SetItemCount(long count)
 {
+    int curr_count = GetItemCount();
+
+    if( curr_count == count )
+        return;
+
+    if( curr_count < count )
+    {
+        for( int i = curr_count; i < count; ++i )
```

Comment:
Isn't this going to be awfully slow? A virtual list control might easily contain a million items or more, if there is any way of inserting them all at once rather than doing it in a loop, we certainly should try to do it.